app/lib/database.py

```diff
@@ -49,99 +48,131 @@ def __init__(self, cursor):
-                file_uuid UUID NOT NULL
-            );
+            SELECT table_name
+            FROM INFORMATION_SCHEMA.views
+            WHERE table_schema = ANY (current_schemas(false))
             """
         )
-        self._cursor.execute(query)
+        result = [ele[0] for ele in self._cursor.fetchall()]
+        return result
 
-    def list_tables(self):
+    def insert_bundle(self, uuid: UUID, version: str, json_as_dict: dict) -> int:
         self._cursor.execute(
             """
-            SELECT table_name
-            FROM information_schema.tables
-            WHERE
-                table_type = 'BASE TABLE' AND
-                table_schema NOT IN ('pg_catalog', 'information_schema') AND
-                table_schema = 'public'
-            """
+            INSERT INTO bundles (uuid, version, json)
+            VALUES (%s, %s, %s)
+            ON CONFLICT (uuid, version) DO NOTHING
+            """,
+            (str(uuid), parse_datetime(version), Json(json_as_dict))
         )
-        result = [ele[0] for ele in self._cursor.fetchall()]
+
+        result = self._cursor.rowcount
         return result
 
-    def insert(self, table_name: str, uuid: UUID, json_as_dict: dict) -> int:
-        query = self._prepare_statement(
+    def insert_metadata_file(self, file_type: str, uuid: UUID, version: str, json_as_dict: dict) -> int:
+        self._cursor.execute(
             """
-            INSERT INTO {} (uuid, json)
-            VALUES (%s, %s)
-            ON CONFLICT (uuid) DO NOTHING
+            INSERT INTO metadata_files (uuid, version, file_type, json)
+            VALUES (%s, %s, %s, %s)
+            ON CONFLICT (uuid, version) DO NOTHING
             """,
-            table_name
+            (str(uuid), parse_datetime(version), file_type, Json(json_as_dict))
         )
-        self._cursor.execute(query, (str(uuid), Json(json_as_dict)))
+        result = self._cursor.rowcount
+        return result
 
+    def insert_join(self, bundle_uuid: UUID, bundle_version: str, file_uuid: UUID, file_version: str) -> int:
+        self._cursor.execute(
+            """
+            INSERT INTO bundles_metadata_files (bundle_uuid, bundle_version, file_uuid, file_version)
+            VALUES (%s, %s, %s, %s)
+            ON CONFLICT (bundle_uuid, bundle_version, file_uuid, file_version) DO NOTHING
+            """,
+            (
+                str(bundle_uuid),
+                parse_datetime(bundle_version),
+                str(file_uuid),
+                parse_datetime(file_version)
+            )
+        )
         result = self._cursor.rowcount
         return result
 
-    def insert_join(self, table_name: str, bundle_uuid: UUID, file_uuid: UUID) -> int:
+    def delete_bundle(self, uuid: UUID, version: str) -> int:
         self._cursor.execute(
             """
-            INSERT INTO bundles_metadata_files
-            VALUES (%s, %s)
+            DELETE FROM bundles
```

Comment:
Should we also delete from the bundles_metadata_files table?

Reply:
After our discussion today, I've decided simply to remove the delete methods.